include/GafferScene/VisibleSet.inl

```diff
@@ -0,0 +1,109 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2022, Cinesite VFX Ltd. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef GAFFERSCENE_VISIBLESET_INL
+#define GAFFERSCENE_VISIBLESET_INL
+
+#include "IECore/PathMatcher.h"
+
+namespace GafferScene
+{
+
+inline PathMatcher::Result VisibleSet::match( const ScenePlug::ScenePath &path, const size_t minimumExpansionDepth ) const
+{
+
+	if( exclusions.match( path ) & ( PathMatcher::ExactMatch | PathMatcher::AncestorMatch ) )
+	{
+		/// Exclusions override all other potential matches for the excluded path and all of its descendants
+		return PathMatcher::Result::NoMatch;
+	}
+
+	unsigned result = PathMatcher::NoMatch;
+	if( minimumExpansionDepth >= path.size() )
+	{
+		result = PathMatcher::ExactMatch;
+	}
+
+	result |= inclusions.match( path );
+	if( result & PathMatcher::AncestorMatch )
+	{
+		/// An ancestor in inclusions will cause this path to be visible so ensure that ExactMatch is also set
+		result |= PathMatcher::ExactMatch;
+	}
+
+	unsigned expansionsMatch = expansions.match( path );
+	if( expansionsMatch & PathMatcher::ExactMatch )
```

Comment:
It shouldn't be necessary for this location to be _expanded_ in order for it to be _visible_ - it's enough for the parent (well, all the ancestors eventually) to be expanded. Here's a test that fails currently :

```
v = GafferScene.VisibleSet()
v.expansions.addPath( "/group" )
assert( v.match( "/group/sphere" ) & ExactMatch
```

Reply:
As above, I think we may need to do this for now to keep the RenderController happy in its current state. I've left a todo in 46e4e68 to revisit this after the RenderController updates.